integration-cli/docker_api_containers_test.go

```diff
@@ -410,9 +410,8 @@ func (s *DockerSuite) TestContainerAPITop(c *testing.T) {
 	if top.Titles[0] != "USER" || top.Titles[10] != "COMMAND" {
 		c.Fatalf("expected `USER` at `Titles[0]` and `COMMAND` at Titles[10]: %v", top.Titles)
 	}
-	assert.Equal(c, len(top.Processes), 2, fmt.Sprintf("expected 2 processes, found %d: %v", len(top.Processes), top.Processes))
-	assert.Equal(c, top.Processes[0][10], "/bin/sh -c top")
```

Comment:
This looks like an optimization in `/bin/sh` -- it could probably be "fixed" by using `"/bin/sh -c 'top && true'"` or similar (to force `/bin/sh` to stick around), if it's really important for this test to have multiple processes.

Reply:
Yes，I have modified the code as you requested. It works.
```
diff --git a/integration-cli/docker_api_containers_test.go b/integration-cli/docker_api_containers_test.go
index 3a30ea8..f6ea1fc 100644
--- a/integration-cli/docker_api_containers_test.go
+++ b/integration-cli/docker_api_containers_test.go
@@ -394,7 +394,7 @@ func (s *DockerSuite) TestContainerAPIPause(c *testing.T) {
 
 func (s *DockerSuite) TestContainerAPITop(c *testing.T) {
        testRequires(c, DaemonIsLinux)
-       out, _ := dockerCmd(c, "run", "-d", "busybox", "/bin/sh", "-c", "top")
+       out, _ := dockerCmd(c, "run", "-d", "busybox", "/bin/sh", "-c", "top && true")
        id := strings.TrimSpace(out)
        assert.NilError(c, waitRun(id))
 
@@ -411,7 +411,7 @@ func (s *DockerSuite) TestContainerAPITop(c *testing.T) {
                c.Fatalf("expected `USER` at `Titles[0]` and `COMMAND` at Titles[10]: %v", top.Titles)
        }
        assert.Equal(c, len(top.Processes), 2, fmt.Sprintf("expected 2 processes, found %d: %v", len(top.Processes), top.Processes))
-       assert.Equal(c, top.Processes[0][10], "/bin/sh -c top")
+       assert.Equal(c, top.Processes[0][10], "/bin/sh -c top && true")
        assert.Equal(c, top.Processes[1][10], "top")
 }
```